wordpress-mobile/WordPress-Android
# Quick start for Existing Users V2 - Media Upload Tour

Parent #16350 

Dependent FluxC PR - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2396

This PR adds the "Upload photos or videos" task to the new `Quick Start` card.

https://user-images.githubusercontent.com/1405144/168525630-ef51ae4f-99ce-44cc-af93-a541c921e509.mp4


https://user-images.githubusercontent.com/1405144/168527305-db9a1feb-89f5-44fa-a548-06c56546c168.mp4


To test:
/cc @jostnes, @osullivanchris 

Test.1 Select "Upload photos or videos" task from the tasks list

1. Launch the app.
2. Enable to `Me` -> `App Settings` -> `Debug settings` -> `QuickStartForExistingUsersV2FeatureConfig`.
3. Restart the app.
4. Logout/login and select "Show me around" when the quick start prompt is shown.
5. Select an existing site.
6. Go to `My Site` tab.
7. ✅ Notice that "Upload photos or videos" task is shown in the new `Quick Start` card with `Get to know the WordPress app` task type.
8. Tap on `Get to know the WordPress app` in the app.
9. Tap on "Upload photos or videos" task. 
10. ✅ Notice that snackbar and qs focus point is displayed to select the media link. (If qs card is shown on Home tab, qs focus point is shown on quick link media item otherwise it is shown on media site item.)
11. Select `media` link.
12. ✅ On the media grid screen, notice that snackbar and qs focus point are displayed to select add new media menu item.
13. Tap on  "+" menu item.
14. Notice that qs focus point  is gone on the add new media menu item.
15. Return to `My Site` tab.
16. ✅ Notice that the complete tasks count is incremented at the bottom of the `Quick Start` card.
17. Tap on`Get to know the WordPress app`.
18. ✅ Notice that "Upload photos or videos" task has moved to the completed tasks list.

Merge Instructions

1. Make sure the corresponding FluxC PR is merged to trunk.
2. Update FluxC version in build.gradle.
3. Remove Not Ready for Merge label
4. Merge the PR.

## Regression Notes
1. Potential unintended areas of impact
New layout added to show quick start focus point on the add new media menu item. Notice that the + button has appropriate color in dark and light mode. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
See "To test" section. Tested in both light and dark mode.

3. What automated tests I added (or what prevented me from doing so)
Media Browser Activity and Media Grid Fragment currently do not follow MVVM pattern to allow writing tests. Refactoring those classes is outside the scope of this PR.

PR submission checklist:
